src/wasm/kotlin/src/arcs/Entity.kt

```diff
@@ -12,38 +12,38 @@ class StringDecoder(private var str: String) {
 
     companion object {
 
-      fun decodeDictionary(str: String): Map<String, String> {
-        val decoder = StringDecoder(str)
-        val dict = mutableMapOf<String, String>()
+        fun decodeDictionary(str: String): Map<String, String> {
+            val decoder = StringDecoder(str)
+            val dict = mutableMapOf<String, String>()
 
-        var num = decoder.getInt(":")
-        while(num-- > 0){
-          val klen = decoder.getInt(":")
-          val key = decoder.chomp(klen)
+            var num = decoder.getInt(":")
+            while(num-- > 0){
+                val klen = decoder.getInt(":")
+                val key = decoder.chomp(klen)
 
-          val vlen = decoder.getInt(":")
-          val value = decoder.chomp(vlen)
+                val vlen = decoder.getInt(":")
+                val value = decoder.chomp(vlen)
 
-          dict[key] = value
-        }
+                dict[key] = value
+            }
 
-        return dict
-      }
+            return dict
+        }
 
       fun decodeList(str: String): List<String> {
-        val decoder = StringDecoder(str)
+          val decoder = StringDecoder(str)
 
-        val list = mutableListOf<String>()
+          val list = mutableListOf<String>()
 
-        var num = decoder.getInt(":")
-        while(num-- > 0) {
-          val len = decoder.getInt(":")
-          val chunk = decoder.chomp(len)
-          list.add(chunk)
-        }
+          var num = decoder.getInt(":")
+            while(num-- > 0) {
```

Comment:
This might even be able to be done with `downTo` and `map`:

```kotlin
list += with(decoder) {
    getInt(":").downTo(0).map { chomp(getInt(":")) }
}
```